### PR TITLE
issue/3098 Fixed inview outofbounds check

### LINKED ISSUE #3098: adapt_framework: inview does not measure out-of-bounds correctly
The `isOutOfBounds` function in `inview.js` assumes that a child will be out-of-bounds in a scrolled container. It is possible to position an element outside of the visible bounds of a parent using many other methods. Checking for container scrolling before checking child offsets is poor logic.

https://github.com/adaptlearning/adapt_framework/blob/833b4ef03760ac78fb5d53eb8b0e009380405bfd/src/core/libraries/inview.js#L543-L573

Example:
When a 3 item narrative is moved to the last item, the middle item is currently considered onscreen. `$('.narrative__slider-image-container[data-index=1]').onscreen().onscreen`

### PR DESCRIPTION
**Pull request by oliverfoster**
fixes #3098 

### Changed
* [Inview out-of-bounds checking to remove scroll pre-check](https://github.com/adaptlearning/adapt_framework/pull/3099/files#diff-53632b9b287dce81b79827993d0cf262f89d524e88fdc845cee2a93880dbf0c2L545-L551)
* [Inview out-of-bounds to drop decimals in comparisons](https://github.com/adaptlearning/adapt_framework/pull/3099/files#diff-53632b9b287dce81b79827993d0cf262f89d524e88fdc845cee2a93880dbf0c2R569-R572)
* Inview linting issues
* [Clarified out-of-bound check by changing the variable language to represent out-of-bound top/left/bottom/right](https://github.com/adaptlearning/adapt_framework/pull/3099/files#diff-53632b9b287dce81b79827993d0cf262f89d524e88fdc845cee2a93880dbf0c2R560-R572)

### CODE REVIEW

File: src/core/libraries/inview.js [hunk 1]
```diff
@@ -542,13 +542,9 @@
...
     isOutOfBounds: function(element, parent) {
 
-      var isScrollWidthOverflowing = (parent.clientWidth < parent.scrollWidth);
-      var isScrollHeightOverflowing = (parent.clientHeight < parent.scrollHeight);
-      var isOverflowing = (isScrollWidthOverflowing || isScrollHeightOverflowing);
-
       var $parent = $(parent);
 
-      if (!isOverflowing || ($parent.css("overflow") === "visible")) {
+      if ($parent.css("overflow") === "visible") {
```
Comment (moloko) [inline src/core/libraries/inview.js:547]:
```suggestion
      if ($parent.css('overflow') === 'visible') {
```

**Review by moloko — APPROVED** [review]
code 👓

**Review by cahirodoherty-learningpool — APPROVED** [review]
👀

**Review by moloko — APPROVED** [review]

**Review by tomgreenfield — APPROVED** [review]
👁️

**Pull request merged**